app/uk/gov/hmrc/contentsecuritypolicyreports/controllers/ReportController.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.contentsecuritypolicyreports.controllers
+
+import org.slf4j.MDC
+import play.api.Logger
+import play.api.libs.functional.syntax.toFunctionalBuilderOps
+import play.api.libs.json._
+import play.api.mvc.{Action, MessagesControllerComponents}
+import uk.gov.hmrc.contentsecuritypolicyreports.controllers.ReportController.ScalaCSPReport
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class ReportController @Inject()(
+  mcc: MessagesControllerComponents,
+) extends FrontendController(mcc) {
+
+  private val cspLogger = Logger("csp-report-logger")
+
+  def report(service: String): Action[JsValue] = {
+    // using tolerantJson since CSP Reports have Content-Type: application/csp-report
+    // we're not using CSPReportBodyParser as it has an issue with String/Long
+    Action(parse.tolerantJson) { implicit request =>
```

Comment:
CLARIFICATION: Could we not just log the payload here rather than parsing it to JSON and then dumping it back out?

Reply:
This is going to be a public endpoint on the internet so we wanted to guard against potentially unwanted content getting into our logs - so we parse/validate json against the expected payload model and only log if it's successful

Reply:
This is also the behaviour of play's CSP Report Body Parser - which we can use once they fix it